com.unity.render-pipelines.core/Runtime/PostProcessing/Shaders/LensFlareHDRPCommon.hlsl

```diff
@@ -0,0 +1,226 @@
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Common.hlsl"
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Filtering.hlsl"
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Random.hlsl"
+#include "Packages/com.unity.render-pipelines.core/ShaderLibrary/Sampling/Sampling.hlsl"
+
+struct AttributesLensFlare
+{
+    uint vertexID : SV_VertexID;
+};
+
+struct VaryingsLensFlare
+{
+    float4 positionCS : SV_POSITION;
+    float2 texcoord : TEXCOORD0;
+    float occlusion : TEXCOORD1;
+};
+
+TEXTURE2D(_FlareTex);
+SAMPLER(sampler_FlareTex);
+
+float4 _FlareColorValue;
+float4 _FlareData0; // x: localCos0, y: localSin0, zw: PositionOffsetXY
+float4 _FlareData1; // x: OcclusionRadius, y: OcclusionSampleCount, z: ScreenPosZ, w: ScreenRatio
+float4 _FlareData2; // xy: ScreenPos, zw: FlareSize
+float4 _FlareData3; // xy: RayOffset, z: invSideCount
+float4 _FlareData4; // x: SDF Roundness, y: SDF Frequency
+float4 _FlareData5; // x: Allow Offscreen, y: Edge Offset, z: Falloff
+
+#define _FlareColor             _FlareColorValue
+
+#define _LocalCos0              _FlareData0.x
+#define _LocalSin0              _FlareData0.y
+#define _PositionOffset         _FlareData0.zw
+
+#define _OcclusionRadius        _FlareData1.x
+#define _OcclusionSampleCount   _FlareData1.y
+#define _ScreenPosZ             _FlareData1.z
+#define _ScreenRatio            _FlareData1.w
+
+#define _ScreenPos              _FlareData2.xy
+#define _FlareSize              _FlareData2.zw
+
+#define _FlareRayOffset         _FlareData3.xy
+#define _FlareShapeInvSide      _FlareData3.z
+
+#define _FlareSDFRoundness      _FlareData4.x
+#define _FlareSDFPolyRadius     _FlareData4.y
+#define _FlareSDFPolyParam0     _FlareData4.z
+#define _FlareSDFPolyParam1     _FlareData4.w
+
+#define _OcclusionOffscreen     _FlareData5.x
+#define _FlareEdgeOffset        _FlareData5.y
+#define _FlareFalloff           _FlareData5.z
+
+float2 Rotate(float2 v, float cos0, float sin0)
+{
+    return float2(v.x * cos0 - v.y * sin0,
+                  v.x * sin0 + v.y * cos0);
+}
+
+#if FLARE_OCCLUSION
+float GetLinearDepthValue(float2 uv)
+{
+#ifdef HDRP_FLARE
```

Comment:
Should this be renamed to something  else other than HDRP since it's in core?

Reply:
For this one it's really HDRP specific.